gnocchi/chef.py

```diff
@@ -54,6 +54,44 @@ def __init__(self, coord, incoming, index, storage):
         self.index = index
         self.storage = storage
 
+    def auto_clean_expired_resources(self, resource_ended_at_normalization):
+        """Cleans expired resources.
+
+        This method will clean resources that have expired according to their 'ended_at' field. The method itself will
+        not execute the cleanup, we will mark the resource as deleted, and leave for the system to execute the actual
+        removal of the data.
+        """
+        moment_now = utils.utcnow()
+        moment = moment_now - datetime.timedelta(seconds=resource_ended_at_normalization)
+        attribute_filter = {"<": {"ended_at": moment}}
+
+        all_resources_found = self.index.list_resources(attribute_filter=attribute_filter)
+        LOG.debug("Resources found for deletion: [%s] that have been expired since [%s].", all_resources_found, moment)
```

Comment:
I think here would be better if you just log the size of the resource list, as you are already logging each resource in the for loop, it would generate some redundant log.

Reply:
Done